### DISCUSSION TITLE
Read Raw M1 and M2 button data

### DISCUSSION THREAD

**Discussion by CallumGracie** [seerge/g-helper#2749] [General]
I am currently using Node-HID to read the data from the inbuilt ROG ALLY Xbox controller and was hoping to read the M1 and M2 buttons but it's my understanding they don't emit on that device. Is there a available HID device where it does and if not where can I read this data from? 

Any help would be appreciated. 

Can provide basic button mappings from node hid if anyone requires them.

**Comment by seerge** [comment]
@CallumGracie hello, M1/M2 don't produce any standard HID commands unless you bind them to emit something specific in AC or G-Helper :)
